### PR TITLE
Restore 2

### PR DESCRIPTION
**Pull request by ChrisTerBeke**
Some refactoring and code cleanup on top of restore_helper.
Basically brings some stuff in line with our code standards, makes some attributes private as they should be, splits some classes, etc.

ping @rvdh I had some spare time, this fixes most of my review comments already.

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
Tests are failing because of coverage, but that can be fixed in the original branch.

File: tests/helpers/TestBaseResourceChecker.py [hunk 1]
```diff
@@ -56,24 +58,24 @@ def test_cleanResources_found(self):
         self.kubernetes_service.getMongoObject.return_value = self.cluster_object
         self.checker.listResources = MagicMock(return_value=[self.cluster_object])
         self.checker.cleanResources()
-        self.assertEqual([call.getMongoObject('mongo-cluster', self.cluster_object.metadata.namespace)],
+        self.assertEqual([call.getMongoObject("mongo-cluster", "mongo-operator-cluster")],
```
Comment (rvdh) [inline tests/helpers/TestBaseResourceChecker.py:61]: Why do we want to hardcode this value?

Comment (ChrisTerBeke) [reply tests/helpers/TestBaseResourceChecker.py:61]: If the variable value changes by accident, we would never find it in the test otherwise.

**Review by rvdh — APPROVED** [review]

**Comment by ChrisTerBeke** [comment]
Feel free to merge at your convenience.

**Comment by rvdh** [comment]
> Feel free to merge at your convenience.

Already done ;)

**Comment by rvdh** [comment]
> Basically brings some stuff in line with our code standards, makes some attributes private as they should be, splits some classes, etc.

Are these code standards documented somewhere?

**Comment by ChrisTerBeke** [comment]
GitHub.com/Ultimaker/Meta. And check the other stardust repos for how we deal with topics like early returns, complexity, separation of concerns, etc

**Pull request merged**